### DISCUSSION TITLE
Transition to `NEWUNIT` option for unique file unit

### DISCUSSION THREAD

**Discussion by scrasmussen** [NCAR/icar#110] [Ideas]
## Transition to `NEWUNIT` option for unique file unit

ICAR currently calls the subroutine `io_newunit` from `io/io_routines.f90` to get a unique file unit. There is a new open option to make sure a file is opened with a unique identifier, e.g. `open(newunit=unit,file='foo.txt')`. I think it would be beneficial to offload that behavior to the compiler implementation.

**Comment by gutmann** [comment]
In general I like the idea, but this was only recently added to some compilers, so requiring it might hurt people running older systems, while not providing any significant benefit that I can see.  Are there reasons this is better? Speed isn't really a concern, the current code has been used a long time and has been robust across compilers / systems, the code doesn't add spaghetti.  

If it ain't broke don't fix it?
